jax/numpy/lax_numpy.py

```diff
@@ -1687,24 +1687,48 @@ def wrapper(*args, **kwargs):
     return asarray(f(*args, **kwargs))
   return wrapper
 
+
+@_wraps(onp.linspace)
 def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
              axis=0):
+  """Implementation of linspace differentiable in start and stop args."""
   lax._check_user_dtype_supported(dtype, "linspace")
-  try:
-    out = onp.linspace(start, stop, num, endpoint, retstep, dtype, axis)
-    if retstep:
-      return asarray(out[0]), out[1]
-    else:
-      return asarray(out)
-  except TypeError:  # Old versions of onp may lack axis arg.
-    out = onp.linspace(start, stop, num, endpoint, retstep, dtype)
-    if retstep:
-      return moveaxis(asarray(out[0]), 0, axis), out[1]
-    else:
-      return moveaxis(asarray(out), 0, axis)
+  dtype = dtype or lax.dtype(start, stop, float(num))
```

Comment:
Can you use `onp.result_type` instead of `lax.dtype` here? `lax.dtype` is meant to be a unary function for getting the dtype of an array or Python scalar (even though its current implementation is just `onp.result_type`).

Reply:
done.